client/blocks/url-preview/index.jsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import { clearPreviewUrl } from 'state/ui/preview/actions';
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
+import { getPreviewUrl } from 'state/ui/preview/selectors';
+import { getSiteOption } from 'state/sites/selectors';
+import addQueryArgs from 'lib/route/add-query-args';
+import { setLayoutFocus } from 'state/ui/layout-focus/actions';
+
+const debug = debugFactory( 'calypso:design-preview' );
+
+export default function urlPreview( WebPreview ) {
+	class UrlPreview extends React.Component {
+		constructor( props ) {
+			super( props );
+			this.state = { previewCount: 0 };
+			this.previewCounter = 0;
+			this.onClosePreview = this.onClosePreview.bind( this );
+			this.getPreviewUrl = this.getPreviewUrl.bind( this );
+			this.getBasePreviewUrl = this.getBasePreviewUrl.bind( this );
+		}
+
+		componentWillReceiveProps( nextProps ) {
+			if ( this.props.selectedSiteId && this.props.selectedSiteId !== nextProps.selectedSiteId ) {
+				this.previewCounter = 0;
```

Comment:
Haha. Ask @ehg. It's a cache-busting mechanism. See 6e872231439a23407f7c68c1a2d28059f2defeba